### PR TITLE
homework-10

### PR DESCRIPTION
**Pull request by Anton-web13**

### CODE REVIEW

File: Homework/js/homework-10.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
+// 1)
+const camelCase = (string) => {
+    const getStringWithSpace = string.split(/(?=[A-Z])/).join(" ");
```
Comment (Boobaley) [inline Homework/js/homework-10.js:3]: I want an alternative solution without googled regexp :)

File: Homework/js/homework-10.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    const maxValue = Math.max(...numbers);
+
+    for (let i = 1; i <= maxValue; i++) {
+        if (i % 2 !== 0 && !numbers.includes(i)) {
+            return i;
+        }
+    }
+
+    return "Nothing is found!"
+};
```
Comment (Boobaley) [inline Homework/js/homework-10.js:44]: your function does't work correct. what if initial data will be [2,4,6,10]?

File: Homework/js/homework-10.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    I: 1,
+    V: 5,
+    X: 10,
+    L: 50,
+    C: 100,
+    D: 500,
+    M: 1000,
+};
+
+const getDecimalNumeral = (allNumeral, date) => {
```
Comment (Boobaley) [inline Homework/js/homework-10.js]: what is date?? There are not any date arg in task

File: Homework/js/homework-10.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    }
+
+    const numberToString = String(decimalCentury);
+
+    if (numberToString.length > 2) {
+        return `${decimalCentury} year`;
+    }
+
+    return `${decimalCentury} century`;
+};
```
Comment (Boobaley) [inline Homework/js/homework-10.js:130]: I suppose you also found this solution on the internet. I want to see yours or explanation how it works.
Actually this is so tricky code which is so hard to read

File: Homework/js/homework-10.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+// MDCLXVI  MMVIII  XXIII  XXI  MMXXII  VIIII
+
+
+// 4)
+const colorsOfGloves = ["red", "green", "red", "blue", "blue"];
+
+const getPairsOfGloves = (colors) => {
+
+    const colorsToObject = colors.reduce((result, number) => {
```
Comment (Boobaley) [inline Homework/js/homework-10.js]: second arg in reduce callback is not a number. is a color

File: Homework/js/homework-10.js [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        return result;
+    }, {});
+
+    const entries = Object.entries(colorsToObject);
+
+    const sumOfPairs = [];
+    let valueOfPairs = 0;
+    for (const value of entries) {
+        const colorOfPairs = value[0];
+        const numberOfPairs = value[1];
```
Comment (Boobaley) [inline Homework/js/homework-10.js]: better to use distructere on for of level

File: Homework/js/homework-10.js [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+        if (numberOfPairs % 2 === 0) {
+            sumOfPairs.push(` Pairs of ${colorOfPairs}: ${numberOfPairs}`);
+            valueOfPairs += numberOfPairs;
+        }
+    }
+
+    return `${valueOfPairs}. ${sumOfPairs}`;
+};
```
Comment (Boobaley) [inline Homework/js/homework-10.js:200]: read the task attentively. 
Your solution works incorrectly

**Review by Boobaley — CHANGES_REQUESTED** [review]

File: Homework/js/homework-10.js (hunk 7 above)
Comment (Anton-web13) [reply Homework/js/homework-10.js:200]: Corrected

File: Homework/js/homework-10.js (hunk 5 above)
Comment (Anton-web13) [reply Homework/js/homework-10.js]: Corrected

File: Homework/js/homework-10.js (hunk 6 above)
Comment (Anton-web13) [reply Homework/js/homework-10.js]: Corrected

File: Homework/js/homework-10.js (hunk 4 above)
Comment (Anton-web13) [reply Homework/js/homework-10.js:130]: No, i dont't found this solution on the internet. It is from me.

File: Homework/js/homework-10.js (hunk 3 above)
Comment (Anton-web13) [reply Homework/js/homework-10.js]: Corrected

**Pull request merged**